lib/node_modules/@stdlib/ndarray/base/unary/test/test.2d.js

```diff
@@ -0,0 +1,1173 @@
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var isSameComplex128Array = require( '@stdlib/assert/is-same-complex128array' );
+var isSameFloat64Array = require( '@stdlib/assert/is-same-float64array' );
+var Complex128Array = require( '@stdlib/array/complex128' );
+var Float64Array = require( '@stdlib/array/float64' );
+var zeros = require( '@stdlib/array/zeros' );
+var oneTo = require( '@stdlib/array/one-to' );
+var ones = require( '@stdlib/array/ones' );
+var Complex128 = require( '@stdlib/complex/float64/ctor' );
+var ndarray = require( '@stdlib/ndarray/ctor' );
+var shape2strides = require( '@stdlib/ndarray/base/shape2strides' );
+var strides2offset = require( '@stdlib/ndarray/base/strides2offset' );
+var numel = require( '@stdlib/ndarray/base/numel' );
+var dfill = require( '@stdlib/blas/ext/base/dfill' );
+var gfill = require( '@stdlib/blas/ext/base/gfill' );
+var blockSize = require( '@stdlib/ndarray/base/unary-tiling-block-size' );
+var unary = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof unary, 'function', 'main export is a function');
+	t.end();
+});
+
+tape( 'the function applies a unary callback to each indexed element of a 2-dimensional ndarray (row-major, singleton dimensions)', function test( t ) {
+	var expected;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+	var y;
+
+	dt = 'float64';
+	ord = 'row-major';
+	sh = [ 4, 1 ];
+	st = shape2strides( sh, ord );
+	o = strides2offset( sh, st );
+
+	x = ndarray( dt, oneTo( numel( sh ), dt ), sh, st, o, ord );
+	y = ndarray( dt, zeros( numel( sh ), dt ), sh, st, o, ord );
+
+	unary( [ x, y ], scale );
+
+	expected = new Float64Array([
+		10.0,
+		20.0,
+		30.0,
+		40.0
+	]);
+	t.strictEqual( isSameFloat64Array( y.data, expected ), true, 'returns expected value' );
+	t.end();
+
+	function scale( x ) {
+		return x * 10.0;
+	}
+});
+
+tape( 'the function applies a unary callback to each indexed element of a 2-dimensional ndarray (row-major, singleton dimensions, accessors)', function test( t ) {
+	var expected;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+	var y;
+
+	dt = 'complex128';
+	ord = 'row-major';
+	sh = [ 4, 1 ];
+	st = shape2strides( sh, ord );
+	o = strides2offset( sh, st );
+
+	x = ndarray( dt, oneTo( numel( sh ), dt ), sh, st, o, ord );
```

Comment:
Ah, right, I don't think you want to use `oneTo` for complex number arrays. Otherwise, all your imaginary components will be zero and you won't be testing whether the implementation is actually appropriately handling complex number arrays. You could do

```suggestion
	x = ndarray( dt, new Complex128Array( oneTo( numel( sh )*2, 'float64' ) ), sh, st, o, ord );
```

Reply:
This same comment will apply here and elsewhere.